app/controllers/user/subscriptions_controller.rb

```diff
@@ -38,15 +38,16 @@ def create
       )
     end
 
-    redirect_to request.referrer
+    redirect_to request.referrer if @user_subscription_form.subscribable.class.name == "Site" ||
```

Comment:
This piece of code is not working well when I follow a Process, because is raising a 500 error:

```
NoMethodError (undefined method `name' for #<GobiertoParticipation::Process:0x007fd7f6ec9ed0>):

app/controllers/user/subscriptions_controller.rb:42:in `create'
```

If you are dealing with HTML and JS requests you should implement a `respond_to`, and probably redirect to the referer in **all** HTML requests, and in the JS requests render a create.js.erb partial. In this partial you should render the button in the new state.

So, when a user follows something the button will say "Proceso seguido!"

Reply:
https://rollbar.com/Populate/gobierto/items/559/

Reply:
This is the mistake I was telling you yesterday that I had pending, I was aware, I correct it and add the test.